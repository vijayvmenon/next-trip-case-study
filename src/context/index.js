import { useContext } from 'react';
import AppContext, { Context } from './context';

export default AppContext;
export { Context };
export const useMyContext = () => useContext(Context);
