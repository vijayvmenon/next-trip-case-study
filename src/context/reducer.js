import constants from '../constants';
import ENTranslations from '../locales/en.json';
import { localStorageExists } from '../utils';

const localStorageLanguage =
  localStorageExists && localStorage.getItem('LANGUAGE');
const initialState = {
  tableExpanded: false,
  hideHeader: false,
  snackbar: { value: '', text: '' },
  language: localStorageLanguage || 'en',
  translations: localStorageLanguage
    ? constants.LANGUAGE_MAPPING[localStorageLanguage]
    : ENTranslations,
};

export default function reducer(state = initialState, action) {
  switch (action.type) {
    case 'TABLE_EXPAND':
      return { ...state, tableExpanded: true };
    case 'TABLE_COLLAPSE':
      return { ...state, tableExpanded: false };
    case 'SET_SNACKBAR':
      return { ...state, snackbar: { value: action.value, text: action.text } };
    case 'HIDE_HEADER':
      return { ...state, hideHeader: true };
    case 'SHOW_HEADER':
      return { ...state, hideHeader: false };
    case 'SET_LANGUAGE':
      return {
        ...state,
        language: action.value,
        translations: constants.LANGUAGE_MAPPING[action.value],
      };
    default:
      return { ...state };
  }
}

export { initialState };
