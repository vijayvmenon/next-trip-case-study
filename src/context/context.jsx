import React, { useReducer, createContext } from 'react';
import PropTypes from 'prop-types';
import reducer, { initialState } from './reducer';

export const Context = createContext();

export default function AppContext({ children }) {
  const [state, dispatch] = useReducer(reducer, initialState);
  return (
    <Context.Provider value={{ state, dispatch }}>{children}</Context.Provider>
  );
}

AppContext.propTypes = {
  children: PropTypes.objectOf(PropTypes.any).isRequired,
};
