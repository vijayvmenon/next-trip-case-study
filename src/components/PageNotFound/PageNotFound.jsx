import React, { useContext, useEffect } from 'react';
import { createUseStyles } from 'react-jss';
import { useLocation } from 'wouter';
import { ReactComponent as AppIcon } from '../../assets/app-logo.svg';
import Button from '../UI/Button';
import { Context } from '../../context';

const useStyles = createUseStyles({
  root: {
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    justifyContent: 'center',
    flexGrow: 1,
  },
  firstLine: {
    display: 'flex',
    alignItems: 'center',
  },
  icon: {
    height: 40,
    width: 40,
    paddingRight: 8,
  },
  notFound404: {
    fontSize: '3rem',
    paddingRight: 20,
  },
  buttonDiv: {
    padding: 16,
  },
  secondLine: {
    padding: 12,
  },
  thirdLine: {
    padding: 12,
  },
});
export default function PageNotFound() {
  const classes = useStyles();
  const { state: { translations = {} } = {}, dispatch } = useContext(Context);
  // eslint-disable-next-line no-unused-vars
  const [location, setLocation] = useLocation();

  useEffect(() => {
    dispatch({ type: 'HIDE_HEADER' });
  }, [dispatch]);

  return (
    <div className={classes.root}>
      <div className={classes.firstLine}>
        {' '}
        <AppIcon className={classes.icon} />
        <span className={classes.notFound404}>404</span>
      </div>
      <div className={classes.secondLine}>{translations.PAGE_NOT_FOUND_1}</div>
      <div className={classes.thirdLine}>{translations.PAGE_NOT_FOUND_2}</div>
      <div className={classes.buttonDiv} data-testid="go-home-button">
        <Button click={() => setLocation('/')} text={translations.GO_HOME} />
      </div>
    </div>
  );
}
