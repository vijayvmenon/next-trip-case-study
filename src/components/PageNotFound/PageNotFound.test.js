import PageNotFound from './PageNotFound';
import renderer from 'react-test-renderer';
import { Context } from '../../context';

const dispatch = jest.fn(); // mock dispatch

describe('Component:PageNotFound.jsx', () => {
  test('PageNotFound.jsx renders correctly', () => {
    const pageNotFound = renderer
      .create(
        <Context.Provider value={{ dispatch }}>
          <PageNotFound />,
        </Context.Provider>,
      )
      .toJSON();
    expect(pageNotFound).toMatchSnapshot();
  });
});
