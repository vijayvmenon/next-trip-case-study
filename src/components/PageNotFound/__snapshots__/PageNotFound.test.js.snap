// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component:PageNotFound.jsx PageNotFound.jsx renders correctly 1`] = `
Array [
  <div
    className="root-0-2-1"
  >
    <div
      className="firstLine-0-2-2"
    >
       
      <svg
        className="icon-0-2-3"
      >
        app-logo.svg
      </svg>
      <span
        className="notFound404-0-2-4"
      >
        404
      </span>
    </div>
    <div
      className="secondLine-0-2-6"
    />
    <div
      className="thirdLine-0-2-7"
    />
    <div
      className="buttonDiv-0-2-5"
      data-testid="go-home-button"
    >
      <button
        className="root-0-2-8"
        data-testid="custom-button"
        onClick={[Function]}
        type="button"
      >
        
      </button>
    </div>
  </div>,
  ",",
]
`;
